M2/Macaulay2/packages/SparseResultants.m2

```diff
@@ -462,6 +463,210 @@ genericMultihomogeneousPolynomial (VisibleList,VisibleList) := o -> (k,d) -> (
     sum(gens A,e -> (b = last baseName e; e * product(n,i -> (mm_i)_(b_i))))
 );
 
+-- Hyperdeterminants --
+
+MultidimensionalMatrix = new Type of HashTable;
+
+MultidimensionalMatrix#{Standard,AfterPrint} = MultidimensionalMatrix#{Standard,AfterNoPrint} = M -> (
+    << endl << concatenate(interpreterDepth:"o") << lineNumber << " : " << "Multidimensional matrix of format " << M#"format" << " over " << M#"ring" << endl;
+);
+
+net MultidimensionalMatrix := M -> net(M#"list");
+
+multidimensionalMatrix = method(TypicalValue => MultidimensionalMatrix);
+
+multidimensionalMatrix (List) := (L) -> (
+    X := makeRing L;
+    F := makeMultilinearForm(L,X);
+    L' := makeList(F,X);
+    new MultidimensionalMatrix from {
+        "format" => apply(X,i -> #i),
+        "list" => L',
+        "multilinearForm" => F,
+        "varsMultilinearForm" => X,
+        "ring" => coefficientRing ring F  
+    }
+);
+
+Thing * MultidimensionalMatrix := (a,M) -> (
+    applyToEntries(M,m -> a*m)
+);
+
+RingElement * MultidimensionalMatrix := (a,M) -> (
+    applyToEntries(M,m -> a*m)
+);
```

Comment:
This method is redundant, since is does the same thing as the more general one above.  It can be removed.

Reply:
Actually the method `Thing * MultidimensionalMatrix` is hidden and undocumented. Otherwise I should also define `ZZ * MultidimensionalMatrix`, etc. The only documented method is `RingElement * MultidimensionalMatrix`.

Reply:
Why not document the actual behavior of your code, so people can understand it better?  When documenting `Thing * MultidimensionalMatrix` you can mention that not all rings are engine rings, and so not all elements of rings inherit from class RingElement.

Reply:
Thinking about it, the method `Thing * Mult..` was not very logical since `Mult.. * Mult..` also exists with a completely different definition. I have added the methods `ZZ * Mult..` and `QQ * Mult..`.